anybox/recipe/odoo/runtime/session.py

```diff
@@ -330,22 +350,28 @@ def update_modules_list(self):
 
         This is necessary prior to install of any new module.
         """
-        self.registry('ir.module.module').update_list(self.cr, self.uid)
+        if hasattr(odoo, 'api'):
+            self.env['ir.module.module'].update_list()
+        else:
+            self.registry('ir.module.module').update_list(self.cr, self.uid)
 
     def init_cursor(self):
         db = getattr(self._registry, 'db', None)
         if db is None:  # current trunk (future v8)
             self.cr = self._registry.cursor()
         else:
-            # In OpenERP < 8, Registry.cursor() object is
+            # In Odoo < 8, Registry.cursor() object is
             # a context manager providing auto closing,
             # but we don't want to control the whole lifespan
             # of the cursor.
             self.cr = db.cursor()
 
     def registry(self, model):
         """Lookup model by name and return a ready-to-work instance."""
-        return self._registry.get(model)
+        if hasattr(odoo, 'api'):
+            return self.env[model]
+        else:
+            return self._registry.get(model)
```

Comment:
Indeed, I think yes!

That make me think, I'll use only the new api in anybox.recipe.odoo as it was said in multiple place earlier we don't expect to keep the compatibility for openerp 7.0 as far we still maintain anybox.recipe.openerp for 7.0 and earlier. Those the code will start to be much clear.

 However the old api must be accessible for this recipe users in upgrade/custom scripts and so on.

I'll do those changes.

Thanks for reply